### ISSUE TITLE
Pretty print table containing hyperlinks for terminal emulators

### ISSUE THREAD

**Issue by crstnbr** [ronisbr/PrettyTables.jl#129]
Some terminals support [printing hyperlinks](https://gist.github.com/egmontkob/eb114294efbcd5adb1944c9f3cb5feda), that is, if the string has a special form, it will be rendered as a clickable link (the url is hidden). This works fine for me using iTerm2. However, if I put the same string into a table and use `pretty_print` the feature is lost. I assume that the format just conflicts with how PrettyTables achieves the formatting.

<img width="994" alt="Screenshot 2021-06-01 at 15 12 16" src="https://user-images.githubusercontent.com/187980/120329074-c4cffc00-c2eb-11eb-89a1-0989ad1e2fb3.png">

Is it completely hopeless to get the best of both worlds or could we perhaps disable the `PrettyTables.jl` pretty-printing for certain table cells?

To give some context, I want to make the JuliaCon talk titles in the following clickable and point to the respective webpages of the talks:

<img width="772" alt="Screenshot 2021-06-01 at 09 24 50" src="https://user-images.githubusercontent.com/187980/120329767-766f2d00-c2ec-11eb-9a53-b7e20a64311e.png">

**Comment by crstnbr** [comment]
(cc'ing @KristofferC who suggested the hyperlink feature in the first place)

**Comment by ronisbr** [comment]
Hi @crstnbr !

Wow, this kind of support is very nice! It is the very first time I am aware of it. I am escaping all characters to avoid problems, but I will check if it is possible to support this. Maybe we need to create a special cell or something.

**Comment by ronisbr** [comment]
Hi @crstnbr !

I need help to test a new system I developed.

The first thing I tried was to natively support this conceal escape sequence. However, it would be **very** difficult to make everything work together with all the options in PrettyTables.jl. Thus, I went a more simple, but limited approach.

I created an abstract type called `CustomTextCell`. If a cell has this type, PrettyTables.jl changes how it is rendered by calling some defined API functions:

- `get_printable_cell_line`: Return the printable text of a line. This must contains only characters that are actually printed (omitting all escape sequences for example).
- `get_rendered_line`: Return the rendered line with all the formatting and escape sequences. This string is what will be written to the buffer.
- `append_suffix_to_line!`: Append a string to the end of the text. This feature is used to add the character `...` when the cell must be cropped.
- `crop_line!`: Tell that the printable text must be cropped by `n` characters.
- `parse_cell_text`: Parse all the lines and return a `Vector{String}` with the printable text.
- `reset!`: Reset all the configurations (cropping, suffix, alignments, etc.).

Those are the activities that PrettyTables.jl must perform to support all the features. The only "downside" is that each structure must add support when parsing the text to options like `autowrap`, `linebreaks`, etc.

Using this API, I created the `URLTextCell` that will provide the feature you want. Hence, we can do:

```Julia
julia> pretty_table(data, alignment = [:r, :l], hlines = [:header], vlines = :none)
 Col. 1  Col. 2
────────────────────────────────────
      1  Google
      2  Apple
      3  Julia
      4  Ronan Arraes Jardim Chagas

julia> data = [1 URLTextCell("Google", "https://www.google.com")
               2 URLTextCell("Apple", "https://www.apple.com")
               3 URLTextCell("Julia", "https://www.julialang.org")
               4 URLTextCell("Ronan Arraes Jardim Chagas", "https://ronanarraes.com")];

julia> header = ["#", "Website"];

julia> pretty_table(data, header = header, alignment = [:r, :l], hlines = [:header], vlines = :none, highlighters = hl_cell(4, 2, crayon"yellow bold"))
```

To create:

<img width="262" alt="Captura de Tela 2021-06-04 às 11 31 50" src="https://user-images.githubusercontent.com/1068295/120817874-75c2da80-c528-11eb-8c87-7bc5b83876b0.png">

This also supports cropping inside the cell and due to the display size:

<img width="197" alt="Captura de Tela 2021-06-04 às 11 32 19" src="https://user-images.githubusercontent.com/1068295/120817945-870be700-c528-11eb-91a1-3bc2c5794390.png">

Can you please test? The code is in the `custom_cell` branch.

**Comment by ronisbr** [comment]
P.S.: This feature (custom text cells) can be used to many very interesting applications. Thus, I appreciate any suggestions about the API :)

**Comment by crstnbr** [comment]
Thanks for the effort! I will test it in a branch for JuliaCon.jl soon!

**Comment by crstnbr** [comment]
Works great! All the links work!

<img width="1460" alt="Screenshot 2021-06-04 at 17 44 22" src="https://user-images.githubusercontent.com/187980/120828200-8725da00-c55c-11eb-82c4-743b7c3119cb.png">

**Comment by ronisbr** [comment]
That's awesome! I will create tests, check for any regressions, commit to `master` and tag a new version so that we can have this during the conference :)

**Comment by crstnbr** [comment]
Would be great if you could tag the new version sooner than later so that I can merge the respective JuliaCon.jl PR etc. Thanks again in advance!

**Comment by ronisbr** [comment]
HI @crstnbr !

Sorry, I am facing some technical problems here 😅 I will be able to tag a new version with this patch until Tuesday.

**Comment by ronisbr** [comment]
Hi @crstnbr 

Just an update. I was about to tag a new version and I discovered a bug when the cell is cropped. It is those kind of bugs that are somewhat difficult to fix :( I would need one more day. I think I will be able to tag this tomorrow. Sorry!

**Comment by crstnbr** [comment]
No worries, just within the next few days would be great! Thanks!

**Comment by ronisbr** [comment]
Hi @crstnbr 

I think everything is ready! Is it possible to test against `master` before I tag the new version? I modified the code a lot.

**Comment by crstnbr** [comment]
Just tested it, looks all great to me!

**Comment by ronisbr** [comment]
Done! It should be available soon (v1.1). Thanks!
